middleware/src/middleware.go

```diff
@@ -21,14 +21,14 @@ type SampleInfo struct {
 type Middleware struct {
 	info        SampleInfo
 	environment system.Environment
-	events      chan *SampleInfo
+	events      chan interface{}
```

Comment:
(Optional) We discussed offline that it might be nice to have a more restricted type for the data flowing through the `middleware` at some point, if we can find or define such a type, since a fully generic `interface{}` tends to lead to a lot of fragile run-time type assertions. Given this goal, it might be good to add a `TODO(TheCharlatan)` describing it here.